WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -94,5 +94,43 @@ public async Task ConfirmConnectionAsync(Guid roundId, Guid aliceId, IEnumerable
 			AmountCredentialClient.HandleResponse(confirmConnectionResponse.ZeroAmountCredentials, zeroAmountCredentialResponseValidation);
 			WeightCredentialClient.HandleResponse(confirmConnectionResponse.ZeroWeightCredentials, zeroWeightCredentialResponseValidation);
 		}
+
+		public async Task SignTransactionAsync(Guid roundId, IEnumerable<ICoin> coinsToSign, BitcoinSecret bitcoinSecret, Transaction unsignedCoinJoin)
+		{
+			if (unsignedCoinJoin.Inputs.Count == 0)
+			{
+				throw new ArgumentException("No inputs to sign.", nameof(unsignedCoinJoin));
+			}
+
+			if (!coinsToSign.Any())
+			{
+				throw new ArgumentException("No coins were provided.", nameof(coinsToSign));
+			}
+
+			var myInputs = coinsToSign.ToDictionary(c => c.Outpoint);
+			var signedCoinJoin = unsignedCoinJoin.Clone();
+			List<InputWitnessPair> signatures = new();
+
+			foreach (var txInput in signedCoinJoin.Inputs.AsIndexedInputs().Where(input => myInputs.ContainsKey(input.PrevOut)))
+			{
+				var coin = myInputs[txInput.PrevOut];
+
+				signedCoinJoin.Sign(bitcoinSecret, coin);
+
+				if (!txInput.VerifyScript(coin, out _))
```

Comment:
I would use the error and log it just in case. Honestly I don't know how useful it could be in production but during development i think it could help in case of some bug.

Reply:
It is kosher, but in this case, it is not elegant because:

- `signedCoinJoin.Sign` does not throw if there is a problem. TransactionBuilder can be a solution, I tried it it is ugly in this case. 
- Additional check is required that is VerifyScript. Do not throw either instead it gives you ScriptError code.

Reply:
I added more comments on errors.